mathics/builtin/importexport.py

```diff
@@ -981,3 +984,61 @@ def apply(self, filename, evaluation):
             return None
 
         return from_python(result)
+
+import base64
+
+class B64Encode(Builtin):
+    """
+    <dl>
+    <dt> 'System`Convert`B64Dump`B64Encode[$expr$]'
+    <dd>Encodes $expr$ in Base64 coding
+    </dl>
+
+    >> System`Convert`B64Dump`B64Encode["Hello world"]
+     = IkhlbGxvIHdvcmxkIg==
+    >> System`Convert`B64Dump`B64Decode[%]
+     = Hello world
+    >> System`Convert`B64Dump`B64Encode[Integrate[f[x],{x,0,2}]]
+     = SW50ZWdyYXRlW2ZbeF0sIHt4LCAwLCAyfV0=
+    >> System`Convert`B64Dump`B64Decode[%]
+     = Integrate[f[x], {x, 0, 2}]
+    """
+    
+    context = "System`Convert`B64Dump`"
+    name = "B64Encode"
+
+    def apply(self, expr, evaluation):
+        'System`Convert`B64Dump`B64Encode[expr_]'
+        if isinstance(expr,String):
+            stringtocodify = "\"" + expr.get_string_value() + "\""
+        else:
+            stringtocodify = Expression('ToString',expr).evaluate(evaluation).get_string_value()
+        return String(base64.b64encode(bytearray(stringtocodify, 'utf8')).decode('utf8'))
+            
+
+
+
+class B64Decode(Builtin):
+    """
+    <dl>
+    <dt> 'System`Convert`B64Dump`B64Decode[$string$]'
+    <dd>Decode  $string$ in Base64 coding to an expression.
+    </dl>
+
+    """
+    
+    context = "System`Convert`B64Dump`"
+    name = "B64Decode"
+
+    def apply(self, expr, evaluation):
+        'System`Convert`B64Dump`B64Decode[expr_String]'
+        try:
+            clearstring = base64.b64decode(bytearray(expr.get_string_value(), 'utf8')).decode('utf8')
+        except Exception as e:
+            print(e.__repr__())
+            return Symbol("$Failed")
+        return Expression('ToExpression', String(six.text_type(clearstring))).evaluate(evaluation)
```

Comment:
In that case, Expression(...) shows the message
Syntax::sntxi: Incomplete expression; more input is needed (line 1 of "")
and returns 
$Failed

Now, I catch this output and in case of fail, an specific message is shown.